hyppo/independence/_utils.py

```diff
@@ -61,6 +63,22 @@ def _check_min_samples(self):
         if nx <= 3 or ny <= 3:
             raise ValueError("Number of samples is too low")
 
+    def _check_redundancy(self):
+        """Check if there are redundancies in either x or y"""
+        u, c = np.unique(self.x, return_counts=True)
```

Comment:
this code is checking for redundant elements in x. #125 mentions redundant rows only for the MGC algorithm

Reply:
So, basically this error check should only be in `MGC.test`

Reply:
So I should put this check_redundancy in the test function in MGC.py script ?

Reply:
After that I should add a test to test_mgc.py to check if the warnings are generated as you gave in the example in the issue.

Reply:
Ok. Thanks

Reply:
> After that I should add a test to test_mgc.py to check if the warnings are generated as you gave in the example in the issue.

yes